lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -304,12 +304,14 @@ $(document).ready(function() {
   }
 
   if( $('.tabnav-tabs').length ){
-    $(".tabnav-tab").click( function() {
+    $(".tabnav-tab").click( function(e) {
+      e.preventDefault();
```

Comment:
Without this `preventDefault`, clicking the Preview tab respects the edit anchor, but then replaces the current anchor with simply `#`. So if you go back to the Edit tab, the section heading is lost. With `preventDefault`, the anchor stays in place, so no matter how often the user switches between the Edit and Preview tab, gollum will automatically move to the selected section. I am not sure which behavior is more desirable, though: if I want to go back to the Edit tab after having edited section `Foo`, do I  thereafter want the editor to jump to section `Foo` continuously? What if I want to edit a different section after `Foo`?

Reply:
> if I want to go back to the Edit tab after having edited section Foo, do I thereafter want the editor to jump to section Foo continuously? What if I want to edit a different section after Foo?

I think one should land roughly where one left it when one switched to the preview pane. Just make sure that the behavior also makes sense for regular, full page editing.